### PR TITLE
add option to use mp3 tones instead of the synthetic

### PR DESCRIPTION
**Pull request by Klaus-Michael**
As I still have issues with the synthetic tones on my Moto G4 Android device I added the option to use mp3 files instead.

### CODE REVIEW

**Comment by pulquero** [comment]
Cool, when running split timing I will be able to use different mp3s for the slave timer to distinguish the enter/exit tones.

**Comment by pulquero** [comment]
Would it be better to introduce high level functions like:

function play_beep(....) {
				if( rotorhazard.use_mp3_tones){
					sound_stage.play();
				}
				else {
					play_beep(100, 440, rotorhazard.tone_volume, 'triangle');
                                }
}

File: src/server/templates/race.html [hunk 1]
```diff
@@ -430,10 +441,15 @@
...
-				setTimeout(function(tone){
-					play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth', );
-				}, 70, msg.node_index);
+				if( rotorhazard.use_mp3_tones){
+					sound_beep.play();
+				}
+				else {
+					play_beep(35, node_tone[msg.node_index], rotorhazard.indicator_beep_volume, 'sawtooth', 0.02);
+					setTimeout(function(tone){
+						play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth', );
```
Comment (pulquero) [inline src/server/templates/race.html]: Is the trailing comma intentional?

File: src/server/templates/race.html [hunk 2]
```diff
@@ -796,14 +822,24 @@
...
-					setTimeout(function(tone){
-						play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth', );
-					}, 70, msg.node_index);
+						if( rotorhazard.use_mp3_tones){
+							sound_exit.play();
+						}
+						else {
+							play_beep(35, node_tone[msg.node_index], rotorhazard.indicator_beep_volume, 'sawtooth', 0.02);
+							setTimeout(function(tone){
+								play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth', );
```
Comment (pulquero) [inline src/server/templates/race.html]: Another trailing comma here.

**Comment by pulquero** [comment]
Not sure if it can be done, but an enhancement (maybe for another PR) would be to dynamically add the audio tags to the DOM so the MP3 files are only loaded when needed.

File: src/server/templates/race.html (hunk 1 above)
Comment (Klaus-Michael) [reply src/server/templates/race.html]: No Idea, thats from the original code done by HazardCreative

Comment (HazardCreative) [reply src/server/templates/race.html]: Can be removed.
```suggestion
						play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth');
```

File: src/server/templates/race.html (hunk 2 above)
Comment (HazardCreative) [reply src/server/templates/race.html]: Can be removed.
```suggestion
								play_beep(35, node_tone[tone], rotorhazard.indicator_beep_volume, 'sawtooth');
```

**Comment by HazardCreative** [comment]
* This is stylistic, but I prefer tones with a fade vs. stopping abruptly. Listen to the current tones on FF and Chrome to hear the difference.
* Switching to MP3 loses the ability to hear which node has activity based on pitch. I do find that valuable.
* High level functions aren't a bad idea, but they'll probably involve a lookup table and have to allow for logic. For example, the crossing exit is a double tone. This is done with two play calls and a delay between them in JS.
* Yes, I would like to see the audio added dynamically. This should be done in JS as a local option and not as a server option, and so that a user can switch without reloading the page.

**Comment by Klaus-Michael** [comment]
- What kind of tones one likes is totally up to personal preference I believe. Basically I simply took the the once I had available from the original delta5 software. I'm a bit used to them and the sound familiar. In the end they can be changed by simply replacing the files.
- Yes there is no longer a way to distinguish between the different nodes, this was a compromise to keep the number of mp3 files to a minimum. Beside this, to be honest, while racing I'm totally unable to distinguish the synthetic tones and map them to a specific node, I assume most people never noticed a difference at all.
- I don't know if high level functions would indeed reduce the code.
- fell free to change this to something dynamic, to be honest I'm not versed enough in js to really make it dynamic.

**Comment by HazardCreative** [comment]
Can you update settings and spectate pages as well?

**Comment by Klaus-Michael** [comment]
is the spectate page in a "stable" state at the moment?
If yes I can add the stuff.

If didn't add anything to the settings page up to now because I wanted this to be client specific. Therefore I didn't see the need for anything on the settings page.
However controlling the "default" setting on the settings page could make sense.

**Comment by HazardCreative** [comment]
Yes, both pages are reasonably stable at the moment.

As long as we have the rest of the audio block in Settings, it should match the Race page. All of the audio settings are client-specific, as noted within the panel.

**Pull request merged**